src/product/product.service.ts

```diff
@@ -70,6 +77,30 @@ export class ProductService {
     });
   }
 
+  async findProductByCategoryId(categoryId: number) {
+    const category = await this.categoryService.findOne(categoryId);
+
+    if (!category) {
+      throw new NotFoundException("Category not found");
+    }
+    const parentId: number = categoryId;
+    const subCategoryId = await this.subCategoryService.findOneSubCategory(parentId);
+
+    if (!subCategoryId) {
+      throw new NotFoundException("SubCategory not found");
+    }
+    const subCategories = await this.subCategoryService.searchSubCategories(parentId);
+    const subCategoryIds: number[] = subCategories.map(subCategory => subCategory.parentId);
+    subCategoryIds.push(parentId);
+
+    const data = await this.prisma.product.findMany({
```

Comment:
In this API we only have to get the product on the basis on category ID so refactor the entire function.